### PR TITLE
Conversation.py update for elevenlabs_extra_body

### PR DESCRIPTION
**Pull request by hikmet-demir**
Update for the extra body, discussed with alex

### CODE REVIEW

File: src/elevenlabs/conversational_ai/conversation.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ class Conversation:
     requires_auth: bool
 
     audio_interface: AudioInterface
+    elevenlabs_extra_body: Optional[dict]
```
[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py]: addressed

File: src/elevenlabs/conversational_ai/conversation.py [hunk 2]
```diff
@@ -161,6 +164,15 @@ def _handle_message(self, message, ws):
             event = message["conversation_initiation_metadata_event"]
             assert self._conversation_id is None
             self._conversation_id = event["conversation_id"]
+            if self.elevenlabs_extra_body:
```
[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py]: addressed

File: src/elevenlabs/conversational_ai/conversation.py [hunk 3]
```diff
@@ -76,6 +77,7 @@ def __init__(
         *,
         requires_auth: bool,
         audio_interface: AudioInterface,
+        elevenlabs_extra_body: Optional[dict] = None,
```
[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py]: addressed

File: src/elevenlabs/conversational_ai/conversation.py [hunk 4]
```diff
@@ -198,7 +227,7 @@ def _handle_message(self, message, ws):
             if self.callback_latency_measurement and event["ping_ms"]:
                 self.callback_latency_measurement(int(event["ping_ms"]))
         else:
-            pass  # Ignore all other message types.
+            print(message["type"])  # Ignore all other message types.
```
[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py:230]: oh yeah let me remove it

File: src/elevenlabs/conversational_ai/conversation.py [hunk 5]
```diff
@@ -137,6 +154,17 @@ def wait_for_session_end(self) -> Optional[str]:
     def _run(self, ws_url: str):
         with connect(ws_url) as ws:
 
+            if self.config:
```
[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py:157]: Yes but I'm getting error for key not found. 

`  File "/Users/hikmet/elevenlabs/Cleaning Docs/elevenlabs-examples/examples/conversational-ai/python/convai/Conversation.py", line 155, in _run
    "custom_llm_extra_body": self.config.extra_body or {},
AttributeError: 'NoneType' object has no attribute 'extra_body'
Conversation ID: None`

[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py:157]: If the customer didn't send any config at all we are failing.

[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py:151]: `                        "custom_llm_extra_body": self.config.extra_body,
                        "conversation_config_override": self.config.conversation_config_override,`
                        
These are failing

[the remark this answers is not in the document]
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py]: okey changedm found a solution

**Comment by hikmet-demir** [comment]
It gives the compile error with the if.self.config.... 
But it works properly on my local without compilation error.

**Comment by hikmet-demir** [comment]
Okey fixed the compile error hopefully

**Comment by hikmet-demir** [comment]
Still compile error, I don't think this CI/CD is correct or I can add the if self.config check back just to pass this

File: src/elevenlabs/conversational_ai/conversation.py [hunk 6]
```diff
@@ -76,6 +85,8 @@ def __init__(
         *,
         requires_auth: bool,
         audio_interface: AudioInterface,
+        config: Optional[ConversationConfig] = ConversationConfig(),
```
Comment (lacop11) [inline src/elevenlabs/conversational_ai/conversation.py]: Optional: This is not modified currently, but would be easy to introduce a bug later by doing so.

https://docs.python.org/3.4/tutorial/controlflow.html#default-argument-values:~:text=Important%20warning%3A%20The%20default%20value%20is%20evaluated%20only%20once.%20This%20makes%20a%20difference%20when%20the%20default%20is%20a%20mutable%20object%20such%20as%20a%20list%2C%20dictionary%2C%20or%20instances%20of%20most%20classes.

Consier `config: Optional[ConversationConfig] = None` and in the body `self.config = config or ConversationConfig()`.

**Review by lacop11 — APPROVED** [review]

File: src/elevenlabs/conversational_ai/conversation.py (hunk 6 above)
Comment (hikmet-demir) [reply src/elevenlabs/conversational_ai/conversation.py]: this is added

**Pull request merged**